### ISSUE TITLE
Dependency from loader is not watched

### ISSUE THREAD

**Issue by just-boris** [webpack/webpack#1458]
I write a loader for my needs. Given a simple code

```
module.exports = function(source) {
    var cb = this.async(),
          self = this;
    someProcessingOperation(source, function(error, result) {
        if(error) {
            return cb(error);
        }
        result.dependencies.forEach(self.addDependency, self);
        cb(null, result.text);
    });
};
```

I expect that my dependencies will be watched and my files will be rebuilt after change. But now it is updated only after changes in source file.

I add to dependencies absolute file paths like this, if it makes sense:

```
/Users/just-boris/projects/project/src/less/blocks/block.less
```

**Comment by just-boris** [comment]
Checked again and found that this doesn't work only when dependency added asynchronously.

If add dependency synchronously, that works fine

**Comment by sokra** [comment]
``` js
result.dependencies.forEach(self.addDependency.bind(self));
```

instead of

``` js
result.dependencies.forEach(self.addDependency);
```

**Comment by just-boris** [comment]
Fixed, and tried again, still not works.
It is expected because `addDependency` is already bound function: https://github.com/webpack/core/blob/master/lib/NormalModuleMixin.js#L106

**Comment by sokra** [comment]
This is an old issue. I guess the problem was solved in the meantime. If not please open a new issue.
